tests/testthat/test-summarize_weighted_production.R

```diff
@@ -175,3 +178,210 @@ test_that("preserves groups passed to ...", {
+
+    expect_error(
+      class = "missing_names",
+      summarize_weighted_percent_change(data, use_credit_limit = use_credit_limit)
+    )
+  }
+
+  expect_error_missing_names("id_loan")
+  expect_error_missing_names("loan_size_outstanding")
+  expect_error_missing_names("loan_size_credit_limit", use_credit_limit = TRUE)
+  expect_error_missing_names("production")
+  expect_error_missing_names("sector")
+  expect_error_missing_names("technology")
+  expect_error_missing_names("year")
+})
+
+test_that("with NAs in crucial columns errors with informative message", {
+  expect_error_crucial_NAs <- function(name, use_credit_limit = FALSE) {
+    data <- fake_master(
+      technology = c("ta", "ta", "tb", "tb"),
+      id_loan = c("i1", "i2", "i1", "i2"),
+      loan_size_outstanding = c(40, 10, 40, 10),
+      production = c(10, 30, 20, 40),
+    )
+
+    data[1, name] <- NA
+    expect_error(
+      class = "some_value_is_missing",
+      summarize_weighted_percent_change(data, use_credit_limit = use_credit_limit)
+    )
+  }
+
+  expect_error_crucial_NAs("id_loan")
+  expect_error_crucial_NAs("loan_size_outstanding")
+  expect_error_crucial_NAs("loan_size_credit_limit", use_credit_limit = TRUE)
+  expect_error_crucial_NAs("production")
+  expect_error_crucial_NAs("sector")
+  expect_error_crucial_NAs("technology")
+  expect_error_crucial_NAs("year")
+})
+
+test_that("with bad but unused loan_size_column is error free", {
+  bad_unused <- rename(fake_master(), bad = "loan_size_credit_limit")
+  expect_error_free(
+    summarize_weighted_percent_change(bad_unused, use_credit_limit = FALSE)
+  )
+
+  bad_unused <- rename(fake_master(), bad = "loan_size_outstanding")
+  expect_error_free(
+    add_weighted_loan_percent_change(bad_unused, use_credit_limit = TRUE)
+  )
+})
+
+test_that("with duplicated loan_size by id_loan throws error", {
+  expect_error(
+    class = "multiple_loan_size_values_by_id_loan",
+    summarize_weighted_percent_change(fake_master(loan_size_outstanding = 1:2))
+  )
+
+  expect_error(
+    class = "multiple_loan_size_values_by_id_loan",
+    summarize_weighted_percent_change(
+      fake_master(loan_size_credit_limit = 1:2),
+      use_credit_limit = TRUE
+    )
+  )
+})
+
+test_that("outputs expected names", {
+  expect_named(
+    summarize_weighted_percent_change(fake_master()),
+    c("sector", "technology", "year", "weighted_percent_change")
+  )
+})
+
+test_that("is sensitive to `use_credit_limit`", {
+  data <- fake_master(
+    name_ald = rep(c("company a", "company b"), 2),
+    year = c(2020, 2020, 2021, 2021),
+    id_loan = c("i1", "i2", "i1", "i2"),
+    loan_size_credit_limit = c(20, 30, 20, 30),
+    production = c(10, 10, 20, 40),
+  )
+  out2 <- data %>%
+    summarize_weighted_percent_change(name_ald, use_credit_limit = TRUE)
+  out3 <- data %>%
+    summarize_weighted_percent_change(name_ald, use_credit_limit = FALSE)
+
+  expect_false(identical(out2, out3))
+})
```

Comment:
This comes from expectations that I found outside a call to `test_that()`. I change the code to reflect how I would test that `summarize_weighted_percent_change()` is sensitive to `use_credit_limit`.

Reply:
Please see also this comment:
https://github.com/2DegreesInvesting/r2dii.analysis/pull/141/commits/df6b90caafd03c54e3cd2f3c6de00520a2c3d2f7#r461865331